bitcoin/src/address.rs

```diff
@@ -554,8 +564,65 @@ impl<'a> fmt::Display for AddressEncoding<'a> {
     }
 }
 
+mod sealed {
+    pub trait NetworkValidation {}
+    impl NetworkValidation for super::NetworkChecked {}
+    impl NetworkValidation for super::NetworkUnchecked {}
+}
+
+/// Marker of status of address's network validation. See section [*Parsing addresses*](Address#parsing-addresses)
+/// on [`Address`] for details.
+pub trait NetworkValidation: sealed::NetworkValidation {}
+
+/// Marker that address's network has been successfully validated. See section [*Parsing addresses*](Address#parsing-addresses)
+/// on [`Address`] for details.
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum NetworkChecked {}
+
+/// Marker that address's network has not yet been validated. See section [*Parsing addresses*](Address#parsing-addresses)
+/// on [`Address`] for details.
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
These enums shouldn't have any derives since they don't do anything but bloat compile times.

Reply:
AFAIK they are needed as long as they are on `Address` struct. If you are OK with implementing the traits on `Address` manually (cf. https://github.com/rust-bitcoin/rust-bitcoin/pull/1489#discussion_r1056758047 ) as follow-up to this PR, I would remove them as part of the follow-up, too.

Otherwise let me know and I will do it as part of this one.

Reply:
Yeah, I meant changing them to manual. I know it's annoying but at least you only have 3-field struct which isn't much. I have a project with several structs each having ~6 fields and a I think at least one impl is for a stupid reason... And I can't hack it with `PhantomData` because I have `T::Assoc`. :man_shrugging: